source/computer/editor.rst

```diff
@@ -0,0 +1,224 @@
+使用 VS Code 编辑远程计算机中的文件了。
+
+文件编辑完毕后，选择菜单栏“文件”中的“关闭远程连接”，或者直接关闭 VS Code，就可以退出远程连接。
+
+.. tip::
+
+   如果需要经常连接远程计算机，可以使用 SSH 配置文件。在活动栏，选择“远程资源管理器”，点击“配置”选择
+   :file:`~/.ssh/config` 配置文件，按照以下格式添加相关信息到该文件中::
+
+       Host seismology
+           User seismo-learn
+           HostName 192.168.1.100
+           IdentityFile ~/.ssh/id_rsa-remote-ssh
+
+   其中，第一列是关键词，如 Host、User、HostName、IdentityFile，第二列是对应的值。seismology 是
+   用户自定义的远程主机标识，其他三个值同上文。可以按照该格式，向该文件中添加多个远程计算机。
+
+   然后点击左下角的远程连接图标，选择“Remote-SSH: Connect to Host”，点击“seismology”就可以登录远程计算机了。
+   也可以通过“远程资源管理器”，点击“seismology”登录远程计算机。
+
+.. tip::
+
+   如果远程计算机使用的 Shell 是 Bash，本地计算机是 Zsh，则可能无法启动 VS Code 的终端，
+   需要修改一下配置文件。打开命令面板，输入 Remote-SSH: Settings 后，搜索 terminal.integrated.shell.linux。
+   然后将 "/bin/zsh" 改为 "/bin/bash" 即可。详情请参考 `microsoft/vscode-remote-release issues #38 <https://github.com/microsoft/vscode-remote-release/issues/38>`__
+
+vim
+----
+
+vim 是一款基于命令行的编辑器，拥有以下特点：
+
+- vim 的快捷键和插件配置，可以让我们使用键盘快速地完成文本编辑。vim 避免了使用鼠标，甚至避免用上下左右键，
+  因为这些需要太多的手指移动
+- vim 轻量级且执行快。对于许多简单任务，启动 vim 比使用图形化编辑器要快地多
+- vim 在很多 Linux 发行版都预装了，某些时候可能是我们唯一能使用的编辑器，比如使用远程计算机
+
+在大多数 Linux 发行版中，vim 也可以用 vi 替代。可以使用以下命令查看::
+
+    $ which vim
+    /usr/bin/vim
+    $ which vi                          
+    vi=vim
+
+本文只介绍 vim 的一些基本用法，可以参考\ `简明 VIM 练级攻略 <https://coolshell.cn/articles/5426.html>`__\
+学习更多用法。
+
+操作模式
+^^^^^^^^^
+
+Vim 的设计以大多数时间都花在阅读、浏览和进行少量编辑改动为基础，因此它具有多种操作模式：
+
+- 正常模式：移动光标、复制、粘贴、删除等
+- 插入模式（INSERT）：插入文本
+- 命令模式：执行一些指令
+- 替换模式（REPLACE）：替换文本
+- 可视化模式：用于选中文本块，分为可视化（VISUAL）、可视化行（VISUAL LINE）、可视化块（VISUAL BLOCK）
+
+在不同的操作模式下，键盘敲击的含义也不同。vim 默认会在左下角显示当前的操作模式。启动时的默认模式是正常模式，
+可以使用以下方式，在不同模式只见进行切换：
+
+- 按下 :kbd:`Esc` 键可以从任何其他模式返回正常模式
+- 在正常模式下，按 :kbd:`i` 键进入插入模式
+- 在正常模式下，按 :kbd:`:` 键进入命令模式
+- 在正常模式下，按 :kbd:`Shift` + :kbd:`R` 键进入替换模式
+- 在正常模式下，按 :kbd:`v` 键进入可视化（一般）模式
+- 在正常模式下，按 :kbd:`Shift` + :kbd:`v` 键进入可视化（行）模式
+- 在正常模式下，按 :kbd:`Ctrl` + :kbd:`v` 键进入可视化（块）模式
+
+日常科研中大部分时间会花在正常模式和插入模式，所有需要用鼠标做的事，都可以用键盘完成。
+
+基本操作
+^^^^^^^^
+
+使用 vim 编辑器打开 :file:`hello-world.txt` 文件，若文件不存在则创建该文件。同时使用以上按键
+在不同的操作模式之间进行切换，并观察左下角地显示::
+
+    # 若 vim 后面没有文件名，也会进入 vim
+    $ vim hello-world.txt
+
+按 :kbd:`i` 键进入插入模式，输入 Hello World，然后按 :kbd:`Esc` 键回到正常模式。
+假设这时文本输入完毕，我们可以进入命令模式（按下 :kbd:`:` 键），这时光标会立即跳到屏幕下方的命令行，
+然后输入 :kbd:`wq` 即可保存文件并退出 vim。
+
+命令模式的常用功能包括保存文件、退出 vim 等：
+
+- :kbd:`:q` 退出 vim 并关闭窗口
+- :kbd:`:w` 保存文件
+- :kbd:`:wq` 保存文件并退出 vim
+
+在正常模式下，可以使用以下常用移动命令在文本中快速移动：
+
+- 基本移动：\ :kbd:`h`\（左）、\ :kbd:`j`\（下）、\ :kbd:`k`\（上）、\ :kbd:`l`\（右）。也可以使用上下左右键
+- 词：\ :kbd:`w`\ （下一个词）、\ :kbd:`b`\ （词初）、\ :kbd:`e`\ （词尾）
+- 行： \ :kbd:`0`\ （行初）、\ :kbd:`^`\ （第一个非空格字符）、\ :kbd:`$`\ （行尾）
+- 屏幕：\ :kbd:`Shift` + :kbd:`h`\ （屏幕首行）、\ :kbd:`Shift` + :kbd:`m`\ （屏幕中间）、\ :kbd:`Shift` + :kbd:`l`\ （屏幕底部）
+- 翻页：\ :kbd:`Ctrl` + :kbd:`u`\ （上翻）、\ :kbd:`Ctrl` + :kbd:`d`\ （下翻）
+- 文件：\ :kbd:`gg`\ （文件头）、\ :kbd:`Shift` + :kbd:`g`\ （文件尾）
+- 行数：{number} + :kbd:`Shift` + :kbd:`g` 会移动至行号为 number（如 20）的行（第 20 行）
```

Comment:
I've never used these keys.

Reply:
Remove some in 870c7ad. I keep three useful move commands.